hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/TransitRegionStateProcedure.java

```diff
@@ -172,11 +172,11 @@ protected boolean waitInitialized(MasterProcedureEnv env) {
 
   private void queueAssign(MasterProcedureEnv env, RegionStateNode regionNode)
       throws ProcedureSuspendedException {
-    // Here the assumption is that, the region must be in CLOSED state, so the region location
-    // will be null. And if we fail to open the region and retry here, the forceNewPlan will be
-    // true, and also we will set the region location to null.
     boolean retain = false;
-    if (!forceNewPlan) {
+    if (forceNewPlan) {
```

Comment:
Have you done any other commit prior to this change? Seems currently TransitRegionStateProcedure does not get called with forceNewPlan = true.

Reply:
Please pull the latest commit. SCP will create TRSP with forceNewPlan = true.